scripts.js

```diff
@@ -0,0 +1,241 @@
+// 1
+function checkIfPropertyIsDefined(object, propertyName) {
+  return object[propertyName] !== undefined;
+}
+
+const user = {
+  firstName: 'John',
+  lastName: 'Smith',
+  age: undefined,
+};
+
+console.log(checkIfPropertyIsDefined(user, 'firstName')); // true
+console.log(checkIfPropertyIsDefined(user, 'height')); // false
+console.log(checkIfPropertyIsDefined(user, 'age')); // false
+
+
+// 2
+const salaries = {
+  john: 50,
+  adam: 100,
+  katie: 75
+}
+const salarySum = salaries.john + salaries.adam + salaries.katie;
+
+
+// 3
+function isObject(value) {
+  return typeof value === 'object' && value !== null;
+}
+
+console.log(isObject({})); // true
+console.log(isObject(10)); // false
+console.log(isObject(null)); // false
+
+// 4
+function checkIfPropertyIsDefined(object, propertyName) {
+  return object[propertyName] !== undefined;
+}
+
+function removeProperty(object, propertyName) {
+
+  if (checkIfPropertyIsDefined(object, propertyName)) {
+    delete object[propertyName];
+    return true;
+  }
+  return false;
+}
+
+const userObject = {
+  name: 'John'
+}
+
+removeProperty(userObject, 'name'); // true
+removeProperty(userObject, 'name'); // false
+removeProperty(userObject, 'weight'); // false
+
+console.log(userObject); // {}
+
+
+// 5
+function checkIfUsersHaveTheSameName(firstObject, secondObject) {
```

Comment:
We know more about those arguments. They are objects that describe users, so we could call them `firstUser` and `secondUser`.